examples/simple/src/main.rs

```diff
@@ -31,7 +31,7 @@ async fn main() {
 
 async fn async_main() {
     info!("Connecting to matchbox");
-    let (mut socket, loop_fut) = WebRtcSocket::new_reliable("ws://localhost:3536/");
+    let (mut socket, loop_fut) = WebRtcSocket::<()>::new_reliable("ws://localhost:3536/");
```

Comment:
This one is particularly weird, the compiler complains it can't infer `C` even thought it doesn't need to

Reply:
I know how to fix it, will send pr

Reply:
oh well, I only did it for ggrs, but the same principle applies for the new_reliable methods; just move them out of the `impl<C>` blocks to an `impl` one.

Reply:
https://github.com/garryod/matchbox/pull/1